chia/_tests/pools/test_pool_cmdline.py

```diff
@@ -1,20 +1,1050 @@
 from __future__ import annotations
 
+import json
+from dataclasses import dataclass
+from typing import Optional, cast
+
 import pytest
-from click.testing import CliRunner
+from chia_rs import G1Element
+
+# TODO: update after resolution in https://github.com/pytest-dev/pytest/issues/7469
+from pytest_mock import MockerFixture
+
+from chia._tests.cmds.cmd_test_utils import TestWalletRpcClient
+from chia._tests.environments.wallet import WalletStateTransition, WalletTestFramework
+from chia._tests.pools.test_pool_rpc import manage_temporary_pool_plot
+from chia._tests.util.misc import Marks, datacases
+from chia.cmds.cmd_classes import NeedsWalletRPC, WalletClientInfo
+from chia.cmds.param_types import CliAddress
+from chia.cmds.plotnft import (
+    ChangePayoutInstructionsPlotNFTCMD,
+    ClaimPlotNFTCMD,
+    CreatePlotNFTCMD,
+    GetLoginLinkCMD,
+    InspectPlotNFTCMD,
+    JoinPlotNFTCMD,
+    LeavePlotNFTCMD,
+    ShowPlotNFTCMD,
+)
+from chia.pools.pool_config import PoolWalletConfig, load_pool_config, update_pool_config
+from chia.pools.pool_wallet_info import PoolSingletonState, PoolWalletInfo
+from chia.rpc.wallet_rpc_client import WalletRpcClient
+from chia.simulator.setup_services import setup_farmer
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.bech32m import encode_puzzle_hash
+from chia.util.errors import CliRpcConnectionError
+from chia.util.ints import uint32, uint64
+from chia.wallet.util.address_type import AddressType
+from chia.wallet.util.wallet_types import WalletType
+from chia.wallet.wallet_state_manager import WalletStateManager
+
+
+@dataclass
+class StateUrlCase:
+    id: str
+    state: str
+    pool_url: Optional[str]
+    expected_error: Optional[str] = None
+    marks: Marks = ()
+
+
+async def verify_pool_state(wallet_rpc: WalletRpcClient, w_id: int, expected_state: PoolSingletonState) -> bool:
+    pw_status: PoolWalletInfo = (await wallet_rpc.pw_status(w_id))[0]
+    return pw_status.current.state == expected_state.value
+
+
+async def process_first_plotnft_create(
+    wallet_test_framework: WalletTestFramework, expected_state: PoolSingletonState
+) -> int:
```

Comment:
Given that these two functions were clearly designed to deduplicate, any interest in deduplicating further by combining them into one and using `len(env.wallet_state_manager.wallets)` instead of hardcoding the IDs or something?  Not a necessity, just might be nice.